dateparser/freshness_date_parser.py

```diff
@@ -111,23 +113,38 @@ def apply_time(dateobj, timeobj):
         return date, period
 
     def _parse_date(self, date_string, prefer_dates_from):
-        if not self._are_all_words_units(date_string):
+
+        _weekday = self.get_weekday_data(date_string)
+
+        if not self._are_all_words_units(date_string) and not _weekday:
             return None, None
 
         kwargs = self.get_kwargs(date_string)
-        if not kwargs:
+
+        if not kwargs and not _weekday:
             return None, None
 
         period = 'day'
-        if 'days' not in kwargs:
-            for k in ['weeks', 'months', 'years']:
-                if k in kwargs:
-                    period = k[:-1]
-                    break
+        if _weekday:
+            day = "cal." + _weekday.upper()
+            day_ahead = eval(day) - self.now.weekday()
```

Comment:
Don’t use `eval`. You should be able to use something else in most scenarios. Here I’m guessing `getattr(cal, _weekday.upper())` would do the job.

Reply:
Done.